wolvic/java/org/chromium/wolvic/HttpAuthManager.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.chromium.wolvic;
+
+import org.chromium.base.annotations.CalledByNative;
+import org.chromium.base.annotations.JNINamespace;
+import org.chromium.base.annotations.NativeMethods;
+import org.chromium.url.GURL;
+
+/**
+ * Manages showing http auth prompt.
+ */
+@JNINamespace("wolvic")
+public final class HttpAuthManager {
+    public interface Bridge {
+        void show();
+        void close();
+    }
+
+    public interface Listener {
+        void proceed(String username, String password);
+        void cancel();
+    }
+
+    public interface Factory {
+        public Bridge create(GURL url, boolean isProxy,
+                             boolean firstAuthAttempt, Listener listener);
+    }
+
+    private static Factory sFactory;
+    private final Bridge mBridge;
+    private long mNativeHttpAuthManager;
```

Comment:
This is a bit of bikeshedding but I'd prefer this to be called `mNativeHttpAuthManagerHandle` for example, otherwise the name suggests it's an object/pointer.

Reply:
Done